src/main/java/edu/kis/powp/jobs2d/drivers/macro/MacroDriver.java

```diff
@@ -0,0 +1,55 @@
+package edu.kis.powp.jobs2d.drivers.macro;
+
+import edu.kis.powp.jobs2d.Job2dDriver;
+import edu.kis.powp.jobs2d.command.DriverCommand;
+import edu.kis.powp.jobs2d.command.OperateToCommand;
+import edu.kis.powp.jobs2d.command.SetPositionCommand;
+import edu.kis.powp.jobs2d.features.DriverFeature;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class MacroDriver implements Job2dDriver {
+    private List<DriverCommand> driverCommandList;
+    private Job2dDriver driver;
+
+    public MacroDriver() {
+        this.driverCommandList = new ArrayList<>();
+    }
+
+    public MacroDriver(List<DriverCommand> driverCommandList) {
+        this.driverCommandList = driverCommandList;
+    }
+
+    public void setDriver(Job2dDriver driver) {
+        this.driver = driver;
+    }
+
+    public void clearCommandSet() {
+        driverCommandList = new ArrayList<>();
+    }
+
+    public List<DriverCommand> getDriverCommandList() {
+        return driverCommandList;
+    }
+
+    public void setDriverFeatureCurrentDriver() {
+        DriverFeature.getDriverManager().setCurrentDriver(driver);
+    }
```

Comment:
A bit complicated logic and dependency with DriverFeature. It would be better to avoid it.